buildenv/jenkins/jobs/infrastructure/artifactory_cleanup.groovy

```diff
@@ -0,0 +1,139 @@
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+*******************************************************************************/
+
+/**
+* This script deletes old artifacts off the Artifactory server 
+* 
+* Parameters: 
+*   JOB_TYPE: Choice - Keep artifacts based on TIME or COUNT
+*   JOB_TO_CHECK: String – The folder to keep x amount of builds in 
+*   ARTIFACTORY_SERVER: String - The artifactory server to clean up
+*   ARTIFACTORY_REPO: String - The artifactory repo to clean up
+*   ARTIFACTORY_NUM_ARTIFACTS: String - How many artifacts to keep in artifactory
+*   ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS: String - Artifacts older than this will be deleted
+**/
+
+timestamps {
+    LABEL = params.LABEL
+    if (!LABEL) {
+        LABEL = 'worker'
+    }
+    node(LABEL) {
+        checkout scm
+        def variableFile = load 'buildenv/jenkins/common/variables-functions'
+        variableFile.parse_variables_file()
+        variableFile.set_artifactory_config()
+
+	    def server = Artifactory.server env.ARTIFACTORY_SERVER
+        def ARTIFACTORY_SERVER_URL = server.getUrl()
+        def ARTIFACTORY_REPO = params.ARTIFACTORY_REPO ? params.ARTIFACTORY_REPO : env.ARTIFACTORY_REPO
+        def artifactoryCreds = env.ARTIFACTORY_CREDS
+        def JENKINS_DOMAIN = params.JENKINS_DOMAIN ? params.JENKINS_DOMAIN : ''
+        def ARTIFACTORY_NUM_ARTIFACTS = params.ARTIFACTORY_NUM_ARTIFACTS ? params.ARTIFACTORY_NUM_ARTIFACTS : env.ARTIFACTORY_NUM_ARTIFACTS
+        def ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS = params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS ? params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS : env.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS
+
+        switch(params.JOB_TYPE) {
+            case 'TIME':
+                cleanupTime(ARTIFACTORY_SERVER_URL, ARTIFACTORY_REPO, ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS, artifactoryCreds)
+            break
+            case 'COUNT':
+                if (params.JOB_TO_CHECK){
+                    cleanupBuilds(ARTIFACTORY_SERVER_URL, ARTIFACTORY_REPO, JENKINS_DOMAIN, params.JOB_TO_CHECK, ARTIFACTORY_NUM_ARTIFACTS, artifactoryCreds)
```

Comment:
Is it possible/easy to add `if (params.JOBS_TO_CHECK == "all")` here and loop though them all? If not, I assume there will be a follow-up PR to call this job in the pipeline builds?

Reply:
My plan was originally making a change like #5338 to call the cleanup script. If I would do `JOBS_TO_CHECK = "all"`, I would need to first get all of the top level folders and I would need to know how at what point I would need to where to count the number of artifacts. I do not believe that all artifacts are located at the same folder depth. I think that It would be simpler if the cleanup script from within a job.